### ISSUE TITLE
Bring back mobile editing for the community edition or at least a cheaper license for this

### ISSUE THREAD

**Issue by arnowelzel** [ONLYOFFICE/DocumentServer#805]
**Do you want to request a *feature* or report a *bug*?**
feature
**What is the current behavior?**
When opening a document in the Nextcloud App on an Android device and the sever runs the community edition of OnlyOffice, a message pops up and says that mobile editing is not possible with the open source version. The same happens when opening a document in a browser which runs in mobile device.

Eventhough I understand, that OnlyOffice also needs to earn some money, paying 950 EUR just to get mobile editing is way too much if you just want to use this as a private person for yourself and your family and friends.
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
Editing documents in the browser or when opening them in the Nextcloud App is possible.
**Did this work in previous versions of DocumentServer?**
Yes
**DocumentServer version:**
5.5.0
**Operating System:**
Ubuntu Linux 16.04 on the server
Android 9 with Nextcloud 3.10.1

**Browser version:**
Firefox 68.5.0 running in Android 9

**Comment by Dextroz** [comment]
This is pretty dirty... Was there any pre-warning of this change? I cannot see anything in the [changelogs](https://github.com/ONLYOFFICE/DocumentServer/blob/master/CHANGELOG.md#550) which references this...

**Comment by Zegorax** [comment]
I second this. This change is really a bad move from OnlyOffice. I had a good trust on you with all the partnership with NextCloud etc, until now.

**Comment by Xeyk** [comment]
Dropped.

Scumbags.

**Comment by immanuelfodor** [comment]
This is real bad news for home usage after such strong partnership announcements with Nextcloud. At least lower the price to 10 bucks or something that people might consider to pay. Or change license to allow unrestricted home usage and make companies pay for theirs.

**Comment by warnerbryce** [comment]
Users get trapped in the honeypot ! 
Now it’s really clear that NC HUb have to drop this opencore project from their product ! 
It will be a major trust crisis for NC

**Comment by YarmoM** [comment]
9 days, no reply from the developers. People/companies need to make money, fair enough. But not this way.

**Comment by warnerbryce** [comment]
@YarmoM 
Step One : Make a nice feature that everybody wants. 
Make a fake level saying no more than 20 connections for having some money on the 50 connections server
Step Two : Give another nice feature for everyone, like mobile edit
Step Three : Take this last feature away for no reason, asking for money because this feature isn't made for the community server... 

Normally on step four you should have a lot of money in your wallet... or the wrath of all a community on you... 
I hope they will change their mind... but i doubt

**Comment by warnerbryce** [comment]
> For people getting a fully-featured enterprise-grade software for free, some of you guys are not very grateful, are you?
> Didn't you consider just for a second that they knew there would be a backlash about this and they went ahead anyway?
> If you need Android apps for editing your documents, there are tons out there (including Microsoft's) so why do you need to use this github collaborative tacking system (also brought to you for free by Microsoft) as a wall of shame?
> Guess what, you are free to change to a competitor, there is no lock-in. Spin up a docker with Collabora On-line and you're good to go (see [ReinerNippes/nextcloud_on_docker#50](https://github.com/ReinerNippes/nextcloud_on_docker/issues/50)). Being a former user of Collabora, I'm happy that Onlyoffice is an alternative, as flawed as it may be.
> BTW, I found the feature description by @arnowelzel to be very polite and a nice ask but the comments just went south from there...

The price for the 50 connections server isn't really high if you're considering having 50 connections. It's less than the Office 365 plan, but if you are 21 users... it's kind of hard to pay for it. 
I rather be willing to pay 10€/month as a donation for having a free software rather than 3,71€/month/per users (if you are 21 users) for a proprietary software that have nothing more than the community version.

**Comment by army1349** [comment]
> 
> 
> For people getting a fully-featured enterprise-grade software for free, some of you guys are not very grateful, are you?

Oh yeah, mobile era mobile-read-only fully-featured enterprise-grade software for free.

**Comment by edo0** [comment]
Scum company, shady as fuck. To be boycotted at all costs

**Comment by YarmoM** [comment]
Can we politely agree that this is about how the decision was presented? Or more specifically, the lack of presentation?

I will reiterate, companies gotta make money, undeniably. And if we got a warning of any sorts, I would have considered my options.

But dropping it this way, making us discover by ourselves that a fundamental feature was taken away, that is the problem. That is not us being ungrateful, that is a company taking its community for granted. Many companies attempt it, few thrive under it.

And you may forgive them on this occasion, but what will they take away next update? What will change once I am fully invested in their ecosystem and dependent on it? These decisions are what makes a company or a project untrustworthy. I personally don't need a mobile editor. I need a reliable project.

A free and decent online office offering without mobile apps is something I am willing to consider (sorry I don't have $950 laying around). But don't treat your users like garbage, especially when the competition is so close.

**Comment by TheManchineel** [comment]
`docker pull onlyoffice/documentserver:5.4.2.46`

**Comment by warnerbryce** [comment]
@TheManchineel Yes, it's nice to have this, but it's still a regression because we won't have updates.
But it's a nice touch before rolling to something else.

**Comment by TheManchineel** [comment]
I wouldn't be suprised if somebody forked this removing the OO name, 20 connections restriction, enabling mobile editing and backporting updates every once in a while into the fork as they come out. Might even cause Ascensio to reconsider the FOSS model, a-la Emby vs. Jellyfin.

**Comment by petrk94** [comment]
> @TheManchineel Yes, it's nice to have this, but it's still a regression because we won't have updates.
> But it's a nice touch before rolling to something else.

honestly, in german schools are still fews which run MS Office 2007, I think I could even live without updates that fast.

**Comment by TheManchineel** [comment]
> 
> 
> > @TheManchineel Yes, it's nice to have this, but it's still a regression because we won't have updates.
> > But it's a nice touch before rolling to something else.
> 
> honestly, in german schools are still fews which run MS Office 2007, I think I could even live without updates that fast.

This. Like, I get it, no updates are bad because security, lack of new features and all, but for people who just need a simple document editor, are they really all that important?

**Comment by e-alfred** [comment]
> Might even cause Ascensio to reconsider the FOSS model

They never were a real FLOSS company, it is just a marketing ploy for them (aka USP) to lure more customers (like "see, we are "Open Source" too"). Their lack of response and communication with the community shows it all.

**Comment by Galegod** [comment]
Hi all, 

Let me try to explain a little bit how and why ONLYOFFICE has not included mobile web version to the latest release of Community Edition Document Server.

First - we didn't want to make things look that way and there was no intention to close the code and make money out of the community. I agree with some of the opinions here - we needed to better cover the situation fo all, really a question to think about, but hope it is never too late. 

We have published Document Server under AGPL v3 and we continue to keep the work completely open and available for all. Web editing for mobile devices is the only feature not included to the latest release. But again - we do have free mobile apps and this is how we plan to move - creating cool application to work with documents on your devices in this mobile era. 
 
All features including mobile web editing are still free for contributors, this is how we understand a fair open source project. Are you ready to contribute in any way - just raise the hand and take the latest version which supports not only viewing in web version but also the editing.
Do not want to invest in ONLYOFFICE and just want to use on your home server - we also have an option, guys, Home Server, and it is just 99$. We have added this option thanks to the issue opened by @arnowelzel as this seems to be a good way to make both sides happy. Do you think it could be a reasonable price for an online office suite (perpetual)?

Then the main question - why we had to exclude the mobile web editing from the Community Edition update: 
that's right, ONLYOFFICE is open source, but at the same time we have to make sure that we are able to continue our work and won't be just thrown out of the market by any IT service provider who starts offering some service based on our project without supporting its further development. It is not just a theory, unfortunately this happens every day as you can see. Also with ONLYOFFICE to my regret.

The feedback we get with the latest events definitely help us to review all the options we can imagine to stay on the stage and continue to work on the online document editors.

**Comment by Zegorax** [comment]
For example, it is impossible to use your native mobile apps with NextCloud / OwnCloud. They require your ONLYOFFICE ERP system.

**Comment by Galegod** [comment]
> For example, it is impossible to use your native mobile apps with NextCloud / OwnCloud. They require your ONLYOFFICE ERP system.

...which is by the way an open source thing under GPL v2

But we do not plan to stop at this point, hope now we have won some time to further development.

**Comment by Zegorax** [comment]
Yes we now it's open-source. But for example, it is nowhere near equivalent in functionalities as Odoo or other systems. Your online editor is great and really good, and the fact that you allow to integrate it in other systems (Nextcloud, Owncloud, etc...) is the best point.

But luring people into having trust in your products, just to remove features without any prior notice is just dishonest and not a great way to build trust for your customers.

**Comment by Galegod** [comment]
Agree, the main mistake was to not tell more in advance, this is what we definitelly need to change

**Comment by Zegorax** [comment]
I would advise you to restore this feature in the next version of OnlyOffice and focusing on new features that maybe won't be available in the newer versions instead.

**Comment by Galegod** [comment]
> Yes we now it's open-source. But for example, it is nowhere near equivalent in functionalities as Odoo or other systems. Your online editor is great and really good, and the fact that you allow to integrate it in other systems (Nextcloud, Owncloud, etc...) is the best point.

And we still completely support a lot of integrations to other systems which can be more known or suitable for the users and add new document storage/erp/mail/crm and other for ONLYOFFICE users https://api.onlyoffice.com/editors/plugins. Every move takes time and resources so we make all possible to make ONLYOFFICE handy.

**Comment by Galegod** [comment]
> Quick edit : You can use them with another cloud provider now. But you cannot use the collaborative feature since the editor is on the device, and not using the one installed at your provider.
Probably this is a good question to the provider....And where we have started with this situation.

**Comment by TheManchineel** [comment]
> Do not want to invest in ONLYOFFICE and just want to use on your home server - we also have an option, guys, Home Server, and it is just 99$.

Yes, thanks. I didn't know it was an option. Does it support Docker & how does one activate it?

**Comment by immanuelfodor** [comment]
> > For example, it is impossible to use your native mobile apps with NextCloud / OwnCloud. They require your ONLYOFFICE ERP system.
> 
> ...which is by the way an open source thing under GPL v2

I self-host a Nextcolud instance, and I wouldn't like to install some random ERP system just to be able to edit a couple of documents on the go a few times in a year. When I need to edit something via mobile, it's like a life-saver that I can/could do it on the web version.

> All features including mobile web editing are still free for contributors, this is how we understand a fair open source project. Are you ready to contribute in any way - just raise the hand and take the latest version which supports not only viewing in web version but also the editing.

Could you please tell more about this option, how could one access the contributor version? This way there would be no need for the previous option.

**Comment by Galegod** [comment]
> > Do not want to invest in ONLYOFFICE and just want to use on your home server - we also have an option, guys, Home Server, and it is just 99$.
> 
> Yes, thanks. I didn't know it was an option. Does it support Docker & how does one activate it?

Sure, it support all the deployment types, also Docker. We have just published this plan - this is why you haven't found it earlier, all the updates take time(
All the steps and guides are available in our helpcenter https://helpcenter.onlyoffice.com/server/integration-edition/index.aspx

**Comment by Zegorax** [comment]
I'm also interested in the contributor edition. Where can we find more information about it ?

**Comment by TheManchineel** [comment]
Wait, so the paid Home server edition actually supports half as many concurrent users as the free (as in freedom and beer) one?

**Comment by Galegod** [comment]
> Could you please tell more about this option, how could one access the contributor version? This way there would be no need for the previous option.

Sure! Just drop a message to documentation@onlyoffice.com - here is more info about how to contribute https://www.onlyoffice.com/contribute.aspx

**Comment by Galegod** [comment]
> I'm also interested in the contributor edition. Where can we find more information about it ?

Here we go https://www.onlyoffice.com/contribute.aspx
Send a mail to documentation@onlyoffice.com

**Comment by Galegod** [comment]
> Wait, so the paid Home server edition actually supports half as many concurrent users as the free (as in freedom and beer) one?

not concurrent users, just named users
Do you think it is not enough for home use?

**Comment by immanuelfodor** [comment]
> > Could you please tell more about this option, how could one access the contributor version? This way there would be no need for the previous option.
> 
> Sure! Just drop a message to [documentation@onlyoffice.com](mailto:documentation@onlyoffice.com) - here is more info about how to contribute https://www.onlyoffice.com/contribute.aspx

Thank you, I've sent an email. This is the most interactive GitHub thread I've ever seen 😀

**Comment by TheManchineel** [comment]
@Galegod Alright, if I understand it correctly, I can share e.g. two spreadsheets and one document with, say, 5 people using a Nextcloud password-protected link (total of 15 connections) and they can all edit collaboratively at the same time, but the same thing can only be done with up to 10 connections if they are accessing directly through Nextcloud accounts? Is this the case?

**Comment by Galegod** [comment]
> This is the most interactive GitHub thread I've ever seen 😀

too many years in support team, sorry)

**Comment by TheManchineel** [comment]
I just learned that GitHub can display new posts in issues without refreshing :smile:

**Comment by Galegod** [comment]
> @Galegod Alright, if I understand it correctly, I can share e.g. two spreadsheets and one document with, say, 5 people using a Nextcloud password-protected link (total of 15 connections) and they can all edit collaboratively at the same time, but the same thing can only be done with up to 10 connections if they are accessing directly through Nextcloud accounts? Is this the case?

not really, only the number of users is limirted, they can use as many devices as they need and open as many documents as they want

**Comment by e-alfred** [comment]
@Galegod You threw your partner Nextcloud under the bus with this move after they integrated your solution into their "Nextcloud Hub" a few weeks ago which took a year to develop according to @jospoortvliet. They implemented Onlyoffice into their mobile clients while it was fully functional and now it is just a viewer anymore.

**Comment by arnowelzel** [comment]
@e-alfred no - OnlyOffice is not implemented in the Nextcloud app, it's just a webview loading the OnlyOffice app from the server. That's the reason why this is only available for Android because in iOS it is not that easy to integrate a webview to an app.

**Comment by immanuelfodor** [comment]
...and it was fantastic while it lasted.

![Screenshot_20200320-181501](https://user-images.githubusercontent.com/21174107/77188797-d8bc8300-6ad6-11ea-94ac-56994d7834e2.jpg)

**Comment by warnerbryce** [comment]
Thank you @arnowelzel 
Now they are trying to rollback this modification. 
They are really happy of this nice move of OnlyOffice. 
The Community is also really happy, it's something Nextcloud really need at the moment.

**Comment by macmiranda** [comment]
> ...and it was fantastic while it lasted.
> 
> ![Screenshot_20200320-181501](https://user-images.githubusercontent.com/21174107/77188797-d8bc8300-6ad6-11ea-94ac-56994d7834e2.jpg)

![Screenshot_20200320_170506_com huawei android internal app](https://user-images.githubusercontent.com/17923899/77189639-2c7b9c00-6ad8-11ea-9c8a-f7cd47464db8.jpg)

**Comment by immanuelfodor** [comment]
> Now they are trying to rollback this modification.
> They are really happy of this nice move of OnlyOffice.
> The Community is also really happy, it's something Nextcloud really need at the moment.

It will be so sad if they'll rollback this webview functionality because of this move. Then those with the contributor or integration version also won't be able to use it. Congrats, we'll finally sit down in the floor between two perfectly usable chairs.

**Comment by arnowelzel** [comment]
@immanuelfodor Nextcloud will not roll back this functionality since it is also used for other apps, like the text editor app of Nextcloud itself (https://apps.nextcloud.com/apps/text) - which also allows live editing with multiple users and may be a good alternative if you don't need a full Office app.

**Comment by warnerbryce** [comment]
Don't worry @immanuelfodor they will rollback with the Edit mode, they have spotlighted the files that were modified.
But i think in some weeks/months, OnlyOffice will make some modification hard coded that won't let Nextcloud use the Mobile Edit with the last updates

**Comment by Zegorax** [comment]
> Don't worry @immanuelfodor they will rollback with the Edit mode, they have spotlighted the files that were modified.
> But i think in some weeks/months, OnlyOffice will make some modification hard coded that won't let Nextcloud use the Mobile Edit with the last updates

The day this happens, it means NextCloud will end their partnership with OnlyOffice.

**Comment by immanuelfodor** [comment]
> @immanuelfodor Nextcloud will not roll back this functionality since it is also used for other apps, like the text editor app of Nextcloud itself (https://apps.nextcloud.com/apps/text) - which also allows live editing with multiple users and may be a good alternative if you don't need a full Office app.

Phew, we've dodged the bullet this time 😃 Thanks for the continuous support!

**Comment by warnerbryce** [comment]
Never put faith in an OpenCore Project.
Nextcloud shake his hand with OnlyOffice and after some days, OnlyOffice stab a knife in his back.
Hey look we have this fresh Home Server for 99$ come here and buy it

**Comment by immanuelfodor** [comment]
It seems the Nextcloud forum is also boiling: https://help.nextcloud.com/t/onlyoffice-removed-web-mobile-editing-from-version-5-5-0-of-community-document-server/74360

**Comment by e-alfred** [comment]
@arnowelzel It is a Webview implementation on the client, but it runs well on both iOS and Android.

**Comment by arnowelzel** [comment]
@e-alfred Thanks for the clarification.

**Comment by Galegod** [comment]
> ...and it was fantastic while it lasted
Did you get the contributors' version already? If the issue is with Nextcloud php version - not sure it can be solved in that way, this is not our version and I doubt we can help there. As soon as you get 
the contribution version you can use ONLYOFFICE Document Server with our connector to Nextcloud without any troubles. And yes, we definitely need to correct the text on the message)

**Comment by Galegod** [comment]
> It will be so sad if they'll rollback the webview functionality because of this move. Then those with the contributor or integration version also won't be able to use it. Congrats, we'll finally sit down on the floor between two perfectly usable chairs.
ONLYOFFICE Document Server was always available for integration to Nextcloud, some years ago we created a connector to use our software together and everyone is still able to use ONLYOFFICE with the latest Nextcloud using this connector.

**Comment by immanuelfodor** [comment]
> > ...and it was fantastic while it lasted
> 
> Did you get the contributors' version already? If the issue is with Nextcloud php version - not sure it can be solved in that way, this is not our version and I doubt we can help there. As soon as you get
> the contribution version you can use ONLYOFFICE Document Server with our connector to Nextcloud without any troubles. And yes, we definitely need to correct the text on the message)

Nope, no reply for the email so far. This is the original version in the NC web view.

**Comment by Galegod** [comment]
> @Galegod You threw your partner Nextcloud under the bus with this move after they integrated your solution into their "Nextcloud Hub" a few weeks ago which took a year to develop according to @jospoortvliet.
Nextcloud shake his hand with OnlyOffice and after some days, OnlyOffice stab a knife in his back.

Not sure I understand your position. What exactly was so incorrect from our side in regard to our partners? We are just working on our own project and source code and have never neglected the rights of other developers or misused someone else's source code.
We completely comply with all the agreements made with the partner but still are a separate project😉

**Comment by immanuelfodor** [comment]
In the meantime, I managed to hide the warning popup, but sadly, a lot more functionality was removed in the original PR, so editing still doesn't work: https://github.com/ONLYOFFICE/web-apps/pull/357  

```bash
# oods is the docker service name in my compose file for onlyoffice documentserver

docker-compose exec oods bash
> grep isSupportEditFeature: /var/www/onlyoffice/documentserver/web-apps/apps/presentationeditor/mobile/app.js
> grep isSupportEditFeature: /var/www/onlyoffice/documentserver/web-apps/apps/documenteditor/mobile/app.js
> grep isSupportEditFeature: /var/www/onlyoffice/documentserver/web-apps/apps/spreadsheeteditor/mobile/app.js
> exit

cat <<EOF > patch-mobile.sh
#!/bin/bash

docker-compose exec oods perl -i -pe 's/(isSupportEditFeature:.+?)\{.+?\}/\1\{return 1\}/' /var/www/onlyoffice/documentserver/web-apps/apps/presentationeditor/mobile/app.js
docker-compose exec oods perl -i -pe 's/(isSupportEditFeature:.+?)\{.+?\}/\1\{return 1\}/' /var/www/onlyoffice/documentserver/web-apps/apps/documenteditor/mobile/app.js
docker-compose exec oods perl -i -pe 's/(isSupportEditFeature:.+?)\{.+?\}/\1\{return 1\}/' /var/www/onlyoffice/documentserver/web-apps/apps/spreadsheeteditor/mobile/app.js
EOF
chmod +x patch-mobile.sh

docker-compose down
docker-compose up -d
./patch-mobile.sh
```

**Comment by Zegorax** [comment]
> In the meantime, I managed to hide the warning popup, but sadly, a lot more functionality was removed in the original PR, so editing still doesn't work: [ONLYOFFICE/web-apps#357](https://github.com/ONLYOFFICE/web-apps/pull/357)
> 
> ```shell
> # oods is the docker service name in my compose file for onlyoffice documentserver
> 
> docker-compose exec oods bash
> > grep isSupportEditFeature: /var/www/onlyoffice/documentserver/web-apps/apps/presentationeditor/mobile/app.js
> > grep isSupportEditFeature: /var/www/onlyoffice/documentserver/web-apps/apps/documenteditor/mobile/app.js
> > grep isSupportEditFeature: /var/www/onlyoffice/documentserver/web-apps/apps/spreadsheeteditor/mobile/app.js
> > exit
> 
> cat <<EOF > patch-mobile.sh
> #!/bin/bash
> 
> docker-compose exec oods perl -i -pe 's/(isSupportEditFeature:.+?)\{.+?\}/\1\{return 1\}/' /var/www/onlyoffice/documentserver/web-apps/apps/presentationeditor/mobile/app.js
> docker-compose exec oods perl -i -pe 's/(isSupportEditFeature:.+?)\{.+?\}/\1\{return 1\}/' /var/www/onlyoffice/documentserver/web-apps/apps/documenteditor/mobile/app.js
> docker-compose exec oods perl -i -pe 's/(isSupportEditFeature:.+?)\{.+?\}/\1\{return 1\}/' /var/www/onlyoffice/documentserver/web-apps/apps/spreadsheeteditor/mobile/app.js
> EOF
> chmod +x patch-mobile.sh
> 
> docker-compose down
> docker-compose up -d
> ./patch-mobile.sh
> ```

I'm on mobile right now so I cannot test but you can try to do the same, but try to replace the IF in the applyLicense here : https://github.com/ONLYOFFICE/web-apps/blob/028c1bd42bcc7e60fd18c79ef67b257f6a5b9a2c/apps/documenteditor/mobile/app/controller/Main.js @ line 644

**Comment by immanuelfodor** [comment]
That IF calls for the isSupportEditFeature function which I made return true, so it's the same at the end as removing the condition completely. It's not just adjusting a boolean, after a look at the original commits, there are lots of code lines removed that supported the edit feature. One example: https://github.com/ONLYOFFICE/web-apps/pull/357/commits/2ab911a439c904a1e1da81283b77639a1f1aa6b1

You either maintain a fork reverting all the changes and merging upstream commits later on, or apply for the contributor version and help the project grow, or buy the $99 integration version as a one-time payment, or buy the $950 version.

I've got back an email from Tatiana (OO) in the meantime, it seems the contributor version is the integration edition with a personal license file that is valid for one year. I also got access to the translation system, I'm looking forward to help the project in return for the continued functionality, a small price to pay to get a better product :)

**Comment by Galegod** [comment]
Thank you @immanuelfodor ! It is more than important for ONLYOFFICE!

**Comment by immanuelfodor** [comment]
> Small edit: the license is lifetime

Are you sure? The received JSON contains the following dates and intervals that look like it's for one year only:
- "end_date":"2021-03-21T23:59:59.000Z"
- "start_date":"2020-03-21T08:06:52.312Z"
- "users_expire":365

In the meantime, I've completed the Hungarian translation in about 2-2.5 hours, no more phrases are missing or need review as of now, a win-win situation for the license! :)

**Comment by arnowelzel** [comment]
About the license: according to https://www.onlyoffice.com/en/integration-edition-prices.aspx the license is lifetime, but you only get updates for one year.

Since the integration in Nextcloud may not stay compatible with the older versions and you usually also want to have security updates this means 99 USD per year (or 149 USD when the promo period is over).

**Comment by immanuelfodor** [comment]
How on Earth could they limit users to one year of updates when the docker builds are openly available? Does it mean it won't be available soon, maybe only from a private registry? Or does the license stop working if you keep updating the docker image after one year, and the image contains an "invalidation" logic? If yes, can it be prevented by clock adjustments to the container? https://stackoverflow.com/questions/29556879/is-it-possible-change-date-in-docker-container

**Comment by immanuelfodor** [comment]
BTW, the contribution edition (integration edition Docker image with the contributor license) indeed gives back the mobile edit feature 😍

![Screenshot_20200321-172356](https://user-images.githubusercontent.com/21174107/77231098-ffdb8900-6b98-11ea-88ef-00b5b5e3fed6.jpg)

**Comment by tnyeanderson** [comment]
Killer damage control guys: [a small admission with a nice price tag attached](https://www.onlyoffice.com/blog/2020/03/integration-edition-at-home-10-users-for-99-per-year/)

Are you going to realize how damaging to your reputation this is? If you want to have separate licenses, then have SEPARATE FEATURES. Develop new features FOR your enterprise products. Don't REMOVE THEM (quietly) from your open edition, then make a sorry blog post only AFTER you are called out on it.

Just put the functionality back in and apologize for this deceptive behavior.... and then STOP being deceptive.

**Comment by Zegorax** [comment]
@tnyeanderson Contact me. I have a solution for you.

**Comment by warnerbryce** [comment]
First-month discount
Home Server costs $149 per year, but during the first month after launch, you can buy it for $99! The discount is valid until April 19.

😂

**Comment by Dextroz** [comment]
Something which I find confusing. On [the announcement page](https://www.onlyoffice.com/blog/2020/03/integration-edition-at-home-10-users-for-99-per-year/) under the what’s included section it says lifetime license but then further down it says per year?

$99 seems a bit steep for home server use but bearable. But $149 for one year is steep imo.

**Comment by e-alfred** [comment]
As @arnowelzel said due to the rapid release cycle of Nextcloud and the fact that this is a public facing service you would have to pay yearly to get updates, otherwise it won't work anymore at one point or somebody would exploit potential security issues.

This makes the term "lifetime" just a useless phrase, because you have to pay subscription-style to keep it working. And after the 99€ (exchange rate of course 1:1) it will cost you 149€ per year.

**Comment by Zegorax** [comment]
I made something yesterday evening that can help you all. You can find it here : https://github.com/Zegorax/OnlyOffice-Unlimited

**Comment by tnyeanderson** [comment]
@Zegorax while I appreciate the hard work and elegant solution, starting a game of licensing cat-and-mouse isn't going to help this situation. Additionally, it wastes the ONLYOFFICE team's resources and gives them someone _else_ to blame. The responsibility needs to continue to be placed squarely on their shoulders, and not shifted back onto the users.

**Comment by Zegorax** [comment]
You are absolutely right and I support your point of view. I made this out of a fun night and it was working quite well at the end so I published it. Bonus point, I'm not even breaking the AGPLv3 license.

**Comment by e-alfred** [comment]
I think in the Nextcloud case, regarding one year of work from @nextcloud integrating Onlyoffice into their "Nextcloud Hub" and giving a huge boost in promotion and ease of installation to Onlyoffice it would be good move to bring back and allow mobile editing inside the Nextcloud clients as a goodwill gesture in the Open Source version.

**Comment by Galegod** [comment]
> I made something yesterday evening that can help you all. You can find it here : https://github.com/Zegorax/OnlyOffice-Unlimited
Bonus point, I'm not even breaking the AGPLv3 license.

@Zegorax Are you sure you understand well enough what you are doing right now? The version you offer the license keys for - is not AGPL v3 Community Edition. 
You have hacked our commercially distributed version - Integration Edition.
Here is the proprietary license agreement link https://onlyo.co/3dniPIq

Hope you correct the situation asap.

**Comment by Zegorax** [comment]
I never agreed to such an agreement. I made this script for https://github.com/ONLYOFFICE/DocumentServer, which is AGPLv3, as described here : https://github.com/ONLYOFFICE/DocumentServer/blob/master/LICENSE.txt

**Comment by Galegod** [comment]
> This makes the term "lifetime" just a useless phrase, because you have to pay subscription-style to keep it working. And after the 99€ (exchange rate of course 1:1) it will cost you 149€ per year.

Sorry for being not clear enough, the contributor version will be always free for all the people who take part in the project.

**Comment by Galegod** [comment]
> I never agreed to such an agreement. I made this script for https://github.com/ONLYOFFICE/DocumentServer, which is AGPLv3, as described here : https://github.com/ONLYOFFICE/DocumentServer/blob/master/LICENSE.txt
> 
> Edit : I do not offer license keys either.

I suggest you check what version you have installed with your script - you can see the ABOUT section with all the references...
You might have known open source versions rarely require any license keys, right?

**Comment by Zegorax** [comment]
license.js (https://github.com/ONLYOFFICE/server/blob/master/Common/sources/license.js) is present in DocumentServer ruled by AGPLv3, open-sourced and present in your repository.

**Comment by Galegod** [comment]
@Zegorax

**Comment by Galegod** [comment]
> license.js (https://github.com/ONLYOFFICE/server/blob/master/Common/sources/license.js) is present in DocumentServer ruled by AGPLv3, open-sourced and present in your repository.

Yes, but you take Integration Edition od Document Server, a commercial product, not the Community Edition, which you mentioned in the link
![image](https://user-images.githubusercontent.com/6768464/77319145-5e327400-6d1f-11ea-81e1-80d841e636ce.png)

**Comment by Zegorax** [comment]
Yes I fixed that, since it was only a comment. If you look at the code of my script, it doesn't install / use / download any of your products. If you want, we can chat about that in the chat of the document you've sent me. (https://onlyo.co/3dniPIq)

**Comment by Galegod** [comment]
> Yes I fixed that, since it was only a comment. If you look at the code of my script, it doesn't install / use / download any of your products. If you want, we can chat about that in the chat of the document you've sent me. (https://onlyo.co/3dniPIq)

You fixed what? A proprietary product? I suggest you think twice since there is nothing about fair compliance with license, just a mistake on your side I hope.

**Comment by tnyeanderson** [comment]
Now the Onlyoffice team has a choice: will they spend their time and resources arguing with Zegorax and trying to mitigate his "solution"? Or will they spend their time actually making NEW features to put behind a paywall, so they can stop taking old ones away?

Seems the former. Shame.

**Comment by Zegorax** [comment]
> You fixed what? A proprietary product? I suggest you think twice since there is nothing about fair compliance with license, just a mistake on your side I hope.

I fixed the usage of "Integration Edition", which was incorrect.

**Comment by Galegod** [comment]
> > You fixed what? A proprietary product? I suggest you think twice since there is nothing about fair compliance with license, just a mistake on your side I hope.
> 
> I fixed the usage of "Integration Edition", which was incorrect. You can even see it in the commit screenshot you've sent earlier.

Integration Edition https://www.onlyoffice.com/ru/integration-edition-prices.aspx

**Comment by Zegorax** [comment]
You do not understand what I'm talking about. I made this script for DocumentServer, which is ruled by AGPLv3. Why are you insisting of Integration Edition when I've already explained that I used this term incorrectly ?

**Comment by arnowelzel** [comment]
Why does the Community Edition need a license key?

**Comment by Zegorax** [comment]
> Why does the Community Edition need a license key?

Community Edition has a maximum of 20 concurrent users without a license

**Comment by tnyeanderson** [comment]
@Galegod no one _wants_ to be creating and using this script right now, I hope you understand that.

You can either listen to your community, or you can fight it.

No one is mad that some features are behind a paywall.

We are mad because the features were removed from the open edition.

We are mad because the features were removed from the open edition _quietly_.

We are mad because the features were removed from the open edition _quietly_, then a "cheaper", "lifetime" (but actually really just one year) license was created to try to control the outrage.

Then you did EXACTLY what I predicted: instead of engaging with the actual issue (the labelled issue at the top of this page), you are arguing with a user who created a script BECAUSE OF WHAT ONLYOFFICE DID.

None of this would have happened if the features hadn't been quietly removed.

Just admit it was a bad move, reverse it, and MOVE ON. Open up an issue on Zegorax/OnlyOffice-Unlimited if you want to discuss that situation, which is off-topic for this thread.

**Comment by Galegod** [comment]
> Community Edition has a maximum of 20 concurrent users without a license

@Zegorax Community Edition like all the software with open source does not require any licenses

**Comment by tnyeanderson** [comment]
@Galegod From the Community Edition link that you posted (still arguing a red herring, please move on from this):


Simultaneous connections | up to 20 maximum
-- | --

**Comment by warnerbryce** [comment]
> > Community Edition has a maximum of 20 concurrent users without a license
> 
Community Edition have a maximum of 20 concurrent users limit hardcoded before compilation. You can modify the file and then compile OnlyOffice to break this limitation

**Comment by Galegod** [comment]
I see and agree that changing anything in an open source project isn't that right as I have already admitted. The fact is we HAD to limit the mobile web version feature since the question is about the future of ONLYOFFICE.

But not for the community! As I have said - we still have the **contribution version for all who want help.**

- Yes, we understand that this is a very risky step.
- Yes, we see that many our users are dissatisfied.
- Yes, we would be more than happy to not take such a step. 

But again - would you prefer ONLYOFFICE to continue to develop a lot of new feautres for document co-editing online, for e2e securing the content of your files, form filling, also a lot of not yet inetgrated features for spreadsheets or presentations, brand new mobile apps and a lot of others (ans yes, some of them has to be paid features, sorry for that)? 
Or is it ok for you that ONLYOFFICE can die just because someone decided to make money out of it without taking into account its future existence?

I hope this does not sound too pathetic and I apologize for the fact that I can not be more specific in why we had to make this really unpopular decision, but I love our project too much to let it go down. 

Again - you are welcome to take the contribution version is it not an option for a community? ONLYOFFICE needs your support, not blaming.
